src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -525,6 +527,7 @@ public CompilationOptions WithOverflowChecks(bool checkOverflow)
         protected abstract CompilationOptions CommonWithCryptoPublicKey(ImmutableArray<byte> cryptoPublicKey);
         protected abstract CompilationOptions CommonWithDelaySign(bool? delaySign);
         protected abstract CompilationOptions CommonWithCheckOverflow(bool checkOverflow);
+        internal abstract CompilationOptions CommonWithMetadataImportOptions(MetadataImportOptions value);
```

Comment:
MetadataImportOptions is internal. Protected would expose it outside of the assembly.